opm/common/OpmLog/MessageFormatter.hpp

```diff
@@ -60,7 +60,12 @@ namespace Opm
         virtual std::string format(const int64_t message_flag, const std::string& message) override
         {
             std::string msg = message;
-            if (use_prefix_) {
```

Comment:
I think the use_prefix_ member (and the constructor argument) is now misleading: for some messages you will get a prefix even if you pass `use_prefix` as `false` to the constructor. I propose the following change:
 - Keep the existing constructor, but replace the bool use_prefix_ member with an int64_t prefix_flag_. The existing constructor initializes prefix_flag_ to the union of all types if use_prefix is true and to 0 if false. That way we keep existing behavior.
 - Add a new constructor that takes the prefix_flag_ directly (and the use_color_coding bool).
 - Add a third constructor taking only the use_color_coding bool, that initializes the prefix_flag_ to the default you want (Warning + Error + Problem + Bug).

That would be both very configurable, keep existing code working as is, and give easy access to the default behavior you want to have.